### DISCUSSION TITLE
DayZ - Mods - Profile folder?

### DISCUSSION THREAD

**Discussion by ghost** [GameServerManagers/LinuxGSM#3899] [Q&A] [answered]
Hi!

Many mods describes the use of a profile folder. After having tried pretty much everything I take my question to te community. Has anyone got this working using mods?

Edit: Ultimately I guess I'm wondering how to get Community Framework working and additional mods.

**Comment by PandaStudiosSWE** [comment, marked as answer]
Yes,

what i did to solve this was to manually create the profiles folder inside of serverfiles folder then i edited dayzserver.cfg inside the lgsm folders

1: Manually create a folder called profiles inside of serverfiles folder

2: go to /home/dayz/lgsm/config-lgsm/dayzserver and edit the dayzserver.cfg file 

what i have in my config looks like this:

```
startparameters="-ip=${ip} -port=${port} -config=${servercfgfullpath} -profiles=profiles '-mod=mods/@SomeRandomMods;mods/SomeMods2;' serverside=${serverside} -bepath=${bepath} -limitFPS=60 -dologs -adminlog -freezeCheck"
```


for me it was important to put the -mod= inside of '' so it looks like this '-mod=' to get it to read all the mods i put in without it only reading the first mod in the list.

I hope this will help you with installing some fun mods.

**Reply by ghost** [reply]
Thanks @PandaStudiosSWE! That did work! I had a bit tunnel vision regarding addig separate parameters as shown in the _default.cfg file. The way of adding "startparameters=" myself had not caught my attention. Nice!

**Reply by JacobOdell** [reply]
What is the absolute path of the "mods" folder in '-mods=mods/@Mod Name1;mods/@Mod Name2;'?

I am trying to get Namalsk to work but it seems it cannot find the mod files. In lgsm/config-lgsm/dayzserver/dayzserver.cfg I have:

```
##################################
####### Instance Settings ########
##################################
# PLACE INSTANCE SETTINGS HERE
## These settings will apply to a specific instance.
startparameters="-config=/home/dayzserver/serverfiles/cfg/dayzserver.server.cfg -profiles=profiles '-mod=mods/@Namalsk Island;mods/@Namalsk Survival' -limitFPS=60 -dologs -adminlog -freezeCheck"
```

But I fail to boot the server when changing the mission to regular.namalsk. In logs I see this: (beginning omitted)

```
...
[S_API FAIL] SteamAPI_Init() failed; SteamAPI_IsSteamRunning() failed.
CAppInfoCacheReadFromDiskThread took 12 milliseconds to initialize
Setting breakpad minidump AppID = 1024020
 1:22:53 SUCCESS: SteamGameServer_Init(0,8766,2302,27016,3,1.18.155069)
 1:22:53 SUCCESS: SteamGameServer_Init(0,8766,2302,27016,3,1.18.155069)
 1:22:53 Server: SetClientState dpid=2 id=0 name=__SERVER__ CREATED -> CONNECTED
DayZ Exp Console version (64bit)1.18.155069 : port 2302
 1:22:53 Server: SetClientState dpid=2 id=0 name=__SERVER__ MISSION SELECTED -> PLAYER ASSIGNED
 1:22:53 Server: SetClientState dpid=2 id=0 name=__SERVER__ PLAYER ASSIGNED -> MISSION RECEIVED
 1:22:53 Roles assigned.
 1:22:53 Reading mission ...
 1:22:53  --- Termination successfully completed ---
No world with name 'namalsk'.
src/tier1/fileio.cpp (5125) : m_vecRegisteredWriters.Count() == 0
src/tier1/fileio.cpp (5125) : m_vecRegisteredWriters.Count() == 0
 1:22:53 Memory should be free now.
 1:22:53   38694280 B not deallocated

```

**Reply by JacobOdell** [reply]
Nevermind, I resolved the issue that I was asking. I am still having the same issue shown in the list, and can verify that the mod is loading (though maybe improperly, it shows up as "$str_nam_mod_surv_name"). There is something else going on.

The path is: /home/dayzserver/serverfiles/mods/@Mod Name1

**Reply by MassCraxx** [reply]
I have the same issue you saw above here, server terminates on startup with no world log. 
`@Namalsk Island` and `@Namalsk Survival` are in the mods folder, `hardcore.namalsk` in mpmissions, cfg adjusted to use hardcore.namalsk and starting server using this dayzserver.cfg:

```startparameters="-ip=${ip} -port=${port} -config=${servercfgfullpath} -profiles=profiles '-mod=mods/@Namalsk Island;mods/@Namalsk Survival;' -limitFPS=60 -dologs -adminlog -freezeCheck"```

but it stops and last log states 

`ErrorMessage: No world with name 'namalsk'.`

@JacobOdell How did you solve this issue?

**Reply by PandaStudiosSWE** [reply]
Hello @MassCraxx @JacobOdell 

Try renaming the mods folder from @Namalsk Survival to @NamalskSurvival since it does not always like spaces in names and do the same for all mods

**Reply by MassCraxx** [reply]
I tried `mods/@Namalsk Survival` - `mods/@NamalskSurvival` - `@NamalskSurvival` with accordingly changing the start params ofc, but no change.

**Reply by JayC1337** [reply]
Old DayZ modding problem which exists since the Arma 2 mod: You have to change every file and folder to lowercase in your mod directory. Easiest way: apt install rename. You can rename files inside a folder with rename 'y/A-Z/a-z/' *

**Reply by MassCraxx** [reply]
Yes! Thank you so much for the tip, with all files and folders in lowercase the server starts up fine. 
Unfortunately, connecting on the client with the same mods selected results in a frozen loading screen. To other servers I can connect just fine. I described my problem in detail [here](https://github.com/GameServerManagers/LinuxGSM/discussions/3980#discussioncomment-3830312). Any further thoughts on that?
